File: ACM/ViveLaDifference/vive.cpp
#include <iostream>

int main() {
  int a, b, c, d;
  
  for (;;) {

    std::cin >> a >> b >> c >> d;
    if (a == 0) break;

    int counter = 0;

    while (a != 0 || b != 0 || c || 0 || d != 0) {

      int a2 = a;

      a = std::abs(a-b);
      b = std::abs(b-c);
      c = std::abs(c-d);
      d = std::abs(d-a2);

      //std::cout << a << " " << b << " " << c << " " << d << std::endl;

      counter++;
    }

    std::cout << counter-1 << std::endl;

  }

  return 0;
}

File: ACM/ViveLaDifference/Vive.py
import math

while(True):
  a, b, c, d = [int(e) for e in raw_input().split()]
  if (a == b == c == d == 0): break
  counter = 0
  while (a != b or b != c or c != d):
    a1, b1, c1, d1 = a, b, c, d
    a = abs(a1-b1)
    b = abs(b1-c1)
    c = abs(c1-d1)
    d = abs(d1-a1)
    #print(str(a) + " " + str(b) + " "+str(c)+" "+str(d))
    counter += 1
  
  print counter
